ConfigurationInitializer/ConfigurationDecryptor.cs

```diff
@@ -44,5 +49,17 @@ private string DecryptBase64StringIfPossible(string base64String)
                 throw;
             }
         }
+        private string DecryptComplexBase64String(string complexBase64String)
+        {           
+            try
+            {
+                return _cryptoProvider.Decrypt(complexBase64String);
+            }
+            catch (CryptographicException ex)
```

Comment:
Is that thrown when the required certificate is missing?
what happens when the regex caught a non encrypted string?